src/ot_tracer.erl

```diff
@@ -31,27 +30,26 @@
 -callback start_span(opentelemetry:tracer(),
                      opentelemetry:span_name(),
                      ot_span:start_opts()) -> opentelemetry:span_ctx().
--callback with_span(opentelemetry:tracer(), opentelemetry:span_ctx()) -> ok.
--callback with_span(opentelemetry:tracer(), opentelemetry:span_ctx(), fun()) -> ok.
--callback end_span(opentelemetry:tracer()) -> ok.
+-callback set_span(opentelemetry:tracer(), opentelemetry:span_ctx()) -> ok.
+-callback end_span(opentelemetry:tracer(), opentelemetry:span_ctx()) -> ok.
 -callback current_span_ctx(opentelemetry:tracer()) -> opentelemetry:span_ctx().
 -callback span_module(opentelemetry:tracer()) -> module().
 
 -spec start_span(opentelemetry:tracer(), opentelemetry:span_name(),
                  ot_span:start_opts()) -> opentelemetry:span_ctx().
 start_span(Tracer={Module, _}, Name, Opts) ->
-    Module:start_span(Tracer, Name, Opts).
-
-with_span(Tracer={Module, _}, Span) when is_atom(Module) ->
-    Module:with_span(Tracer, Span).
-
--spec with_span(opentelemetry:tracer(), opentelemetry:span_ctx(), fun()) -> ok.
-with_span(Tracer={Module, _}, SpanCtx, Fun) when is_atom(Module) ->
-    Module:with_value(Tracer, SpanCtx, Fun).
+    SpanCtx = Module:start_span(Tracer, Name, Opts),
+    Module:set_span(Tracer, SpanCtx),
```

Comment:
Stuff like this shouldn't be in the API. It is up to the SDK.

Reply:
I tend to disagree. Mostly because this is a thing that **all** tracer modules will need to do anyway. There is no way to escape this. For me moving it single layer above makes job simpler for implementors who will not need to think about setting proper variables on their own. It will simply reduce the cognitive overhead and will make tracer module much simpler. I would even go and rename callback functions `start_span` -> `create_span` and `end_span` -> `store_span`.

Reply:
Na, it isn't, at least not in the current spec. It has been discussed, and the API is not meant to do this, an SDK could never set a span to the context.

Reply:
If I [read this](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#span-creation) correctly then neither should SDK set it as an active span. We only should expose `set_span/2` function as well.

Strange decision to make IMHO, but ok.